app/controllers/comments_controller.rb

```diff
@@ -3,26 +3,22 @@ class CommentsController < ApplicationController
   http_basic_authenticate_with name: "dhh", password: "secret"
 
   def create
-    @article = Article.find(params[:article_id])
-    @comment = @article.comments.build(comment_params)
-
-    if @comment.save
-      redirect_to @article
+    @comment = Comments::Create.call(comment_params)
+    if @comment.errors.any?
+      render Article.find(comment_params[:article_id]), status: :unprocessable_entity
```

Comment:
`render Article.find(comment_params[:article_id])` looks ugly and not necessary. Use named route instead.